### PR TITLE
feat: Setup integration tests workflow

### PR DESCRIPTION
**Pull request by dantehemerson**
* Setup Postgres docker image to run integration tests
* Use [supertest](https://github.com/visionmedia/supertest) to test http services
* Change tabs to spaces(2)

### CODE REVIEW

File: src/__tests__/app.int-spec.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import request from 'supertest';
+import * as faker from 'faker';
+import { HttpStatus } from '@shared/enums/http-status.enum';
+import { App } from '../app';
+import { aUUID, aISODate } from './helpers/matchers.helper';
+
+jest.mock('../shared/infra/typeorm');
```
Comment (dantehemerson) [inline src/__tests__/app.int-spec.ts:7]: Use 
`src/shared/infra/typeorm/__mocks__/index.ts`
instead
`src/shared/infra/typeorm/index.ts`

File: src/__tests__/app.int-spec.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+				return request(app.getServer())
+					.post('/users')
+					.send({
+						name: 'Dante',
+						email: 'naguadtl@gmail.com',
+						password: 'dante',
+					})
+					.expect('Content-Type', /json/)
+					.expect(HttpStatus.CREATED)
+					.expect(({ body }) => {
```
Comment (dantehemerson) [inline src/__tests__/app.int-spec.ts:45]: @stevescruz The code identation looks weird, we are using 2 for `indent_size` in `.editorconfig` and we are using tabs for identation. Maybe we can use spaces?

File: .eslintrc.json [hunk 3]
```diff
@@ -65,6 +65,7 @@
 			}
 		],
 		"import/prefer-default-export": "off",
+		"import/no-extraneous-dependencies": ["error", { "devDependencies": true }],
```
Comment (dantehemerson) [inline .eslintrc.json:68]: In order to use testing libraries without warning.

File: jest.setup.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const { execSync } = require('child_process');
+
+module.exports = function () {
+	console.log('\nSetup Docker for testing');
+	const stdout = execSync(
+		'docker-compose -p docker_app_test -f docker-compose.test.yml up -d',
+	);
+	console.log(stdout.toString('utf-8'));
+};
```
Comment (dantehemerson) [inline jest.setup.js:9]: Setup Postgres image(docker-compose.test.yml) for integration tests.
`jest.setup.js` and `jest.teardown.js` are executed on start running tests and finish running tests respectively.

File: src/__tests__/app.int-spec.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+						name: 'Update',
+					})
+					.expect('Content-Type', /json/)
+					.expect(HttpStatus.NOT_FOUND)
+					.expect(({ body }) => {
+						expect(body).toMatchObject({
+							message: 'User not found',
+							name: 'NotFoundError',
+						});
+					});
```
Comment (dantehemerson) [inline src/__tests__/app.int-spec.ts:72]: This is an example of Integration tests. Testing update user service.

File: src/__tests__/app.int-spec.ts (hunk 2 above)
Comment (stevescruz) [reply src/__tests__/app.int-spec.ts:45]: Yeah, it looks really weird on your example.

We can use spaces, no problem

**Review by stevescruz — APPROVED** [review]

**Pull request merged**